pkg/store/bucket.go

```diff
@@ -891,6 +926,13 @@ func (s *BucketStore) Series(req *storepb.SeriesRequest, srv storepb.Store_Serie
 
 	s.mtx.RLock()
 
+	var ts []tombstone.Tombstone
+	//Loading all tombstones from objstore
+	ts, err = tombstone.ReadTombstones(s.bkt, s.logger)
```

Comment:
Performance wise this could be impactful. What if the bucket store keeps a background process to sync tombstones and keeping them in memory, so that at query time all tombstones are already loaded in memory without blocking to read them?

Reply:
Maybe we could use https://github.com/thanos-io/thanos/blob/c5dd131a422f58b110463a9bb5766eb654bcf3f4/pkg/store/cache/caching_bucket_factory.go#L67

Reply:
Yes, would be an improvement. However, tombstones look to be a very good candidate for async scanning (especially when there are a bunch of them).